File: test2.js
let products=[
   {"name":"Product A","price":100,"stock":5},
   {"name":"Product B","price":200,"stock":3} ,
   {"name":"Product C","price":50,"stock":10}
]

let sort_key="name";
let ascending=false;

products.sort(function(a,b){
   if(sort_key=="name"){
      if(ascending==true){
         return a[sort_key].localeCompare(b[sort_key]);
      }
      else{
         return b[sort_key].localeCompare(a[sort_key]);
      }
      
   }
   else{
      if(ascending==true){
         return a[sort_key]-b[sort_key];
      }
      else{
         return b[sort_key]-a[sort_key];
      }
   }
})
console.log("SORT LIST",products);

File: test2.py

def get_sorted_list():
   products=[
      {"name":"Product A","price":100,"stock":5},
      {"name":"Product B","price":200,"stock":3} ,
      {"name":"Product C","price":50,"stock":10}
   ]
   sort_key=""
   
   #####Conditionals#####
   sort_key="name"
   ascending=False
   ##################
   
   a=sorted(products, key=lambda d: d[sort_key],reverse=ascending)
   return a

if __name__=="__main__":
   print(get_sorted_list())